### PR TITLE
[CARBONDATA-4026] Fix Thread leakage while Loading

### PR DESCRIPTION
**Pull request by marchpure**
### Why is this PR needed?
 A few code of Inserting/Loading/InsertStage/IndexServer won't shutdown executorservice. leads to thread leakage which will degrade the performance of the driver and executor.
 
 ### What changes were proposed in this PR?
Shutdown executorservices as soon as finish using them.
    
 ### Does this PR introduce any user interface change?
 - No

 ### Is any new testcase added?
 - No

### CODE REVIEW

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2606/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4356/

File: core/src/main/java/org/apache/carbondata/core/statusmanager/StageInputCollector.java [hunk 1]
```diff
@@ -65,8 +65,15 @@
...
-      return createInputSplits(executorService, stageInputFiles);
+      ExecutorService executorService = null;
+      try {
+        executorService = Executors.newFixedThreadPool(numThreads);
+        return createInputSplits(executorService, stageInputFiles);
+      } finally {
+        if (executorService != null && !executorService.isShutdown()) {
+          executorService.shutdownNow();
+        }
+      }
```
Comment (Kejian-Li) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/StageInputCollector.java:76]: executorService has already been shut down, and then pass it in createInputSplits, is that okay?

Comment (Kejian-Li) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/StageInputCollector.java:76]: executorService has already been shut down, and then pass it in createInputSplits, is that okay?

**Comment by Kejian-Li** [comment]
LGTM

File: core/src/main/java/org/apache/carbondata/core/statusmanager/StageInputCollector.java [hunk 2]
```diff
@@ -65,8 +65,15 @@
     collectStageFiles(table, hadoopConf, stageInputFiles, successFiles);
     if (stageInputFiles.size() > 0) {
       int numThreads = Math.min(Math.max(stageInputFiles.size(), 1), 10);
-      ExecutorService executorService = Executors.newFixedThreadPool(numThreads);
-      return createInputSplits(executorService, stageInputFiles);
+      ExecutorService executorService = null;
+      try {
+        executorService = Executors.newFixedThreadPool(numThreads);
```
Comment (QiangCai) [inline core/src/main/java/org/apache/carbondata/core/statusmanager/StageInputCollector.java:70]: move line 70 to line 68

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.3.4, Please check CI http://121.244.95.60:12545/job/ApacheCarbonPRBuilder2.3/4361/

**Comment by CarbonDataQA1** [comment]
Build Success with Spark 2.4.5, Please check CI http://121.244.95.60:12545/job/ApacheCarbon_PR_Builder_2.4.5/2611/

File: core/src/main/java/org/apache/carbondata/core/statusmanager/StageInputCollector.java (hunk 2 above)
Comment (marchpure) [reply core/src/main/java/org/apache/carbondata/core/statusmanager/StageInputCollector.java]: I have modified code according to your suggestion

File: integration/spark/src/main/scala/org/apache/spark/rdd/CarbonMergeFilesRDD.scala [hunk 3]
```diff
@@ -158,20 +158,27 @@ object CarbonMergeFilesRDD {
...
-          executorService.submit(new Runnable {
-            override def run(): Unit = {
-              ThreadLocalSessionInfo.setCarbonSessionInfo(carbonSessionInfo)
-              FileFactory.deleteAllCarbonFilesOfDir(
-                FileFactory.getCarbonFile(partitionPath + "/" + tempFolderPath))
-            }
-          })
+      var executorService: ExecutorService = null
+      try {
+        executorService = Executors.newFixedThreadPool(numThreads)
```
[the remark this answers is not in the document]
Comment (marchpure) [reply integration/spark/src/main/scala/org/apache/spark/rdd/CarbonMergeFilesRDD.scala]: I have modified code according to your suggestion

File: integration/spark/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala [hunk 4]
```diff
@@ -123,6 +123,9 @@ object IndexServer extends ServerInterface {
         t
       }
     })
+    indexServerExecutorService.get.shutdown()
```
[the remark this answers is not in the document]
Comment (marchpure) [reply integration/spark/src/main/scala/org/apache/carbondata/indexserver/IndexServer.scala]: I have modified code according to your suggestion

**Comment by QiangCai** [comment]
LGTM

**Pull request closed without merging**